x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list.tsx

```diff
@@ -1087,45 +660,46 @@ export const RulesList = ({
     showToast({ action: 'DISABLE', errors, total });
     await refreshRules();
     onClearSelection();
-  }, [http, selectedIds, getFilter, setIsDisablingRules, showToast]);
+  };
 
   const onDeleteCancel = () => {
     setIsDeleteModalVisibility(false);
-    clearRulesToDelete();
+    clearRulesToBulkEdit();
   };
-
-  const onDeleteConfirm = useCallback(async () => {
+  const onDeleteConfirm = async () => {
+    if (bulkEditAction !== 'delete') {
+      return;
+    }
     setIsDeleteModalVisibility(false);
-    setIsDeletingRules(true);
```

Comment:
I'll put a little bit more explanation here.
Why `updateRulesToBulkEdit` has delete action, but do not have enable and disable.
`
updateRulesToBulkEdit({
                  action: 'delete',
                  rules: [rule],
                })
`
So you unified all bulk edit functions and delete, name it edit, but did not include disable/enable.

Reply:
Ultimately we want to unify all bulk actions into 1 state. Enable/disable is not here because this PR was created before the bulk enable/disable PR you made was merged.

@guskovaue could you checkout this branch and test your bulk enable/disable to ensure it is still working as expected? If so, we can create another issue to refactor it to add enable/disable to the bulk edits

Reply:
@JiaweiWu I've checked it during my code review. It's working as aspected.
Agree with creating a new issue.

Reply:
I've created a ticket for that: `https://github.com/elastic/kibana/issues/148611`